src/index.ts

```diff
@@ -26,148 +26,148 @@ const tbodyListForAttendance = document.getElementById("tbodyListForAttendance")
 let selectedStudent: Student;
 showLists();
 
-addStudentButton.addEventListener("click", (e) => {
-    console.log("Ekleme");
-    studentService.addStudent(nameInput.value, surnameInput.value);
-    nameInput.value = "";
-    surnameInput.value = "";
-    showLists();
-    e.preventDefault();
+addStudentButton.addEventListener("click", async (e) => {
+  console.log("Ekleme");
+  await studentService.addStudent(nameInput.value, surnameInput.value);
+  nameInput.value = "";
+  surnameInput.value = "";
+  showLists();
+  e.preventDefault();
 });
 
-updateStudentButton.addEventListener("click", (e) => {
-    studentService.updateStudent(selectedStudent.id, nameInput.value, surnameInput.value);
-    nameInput.value = "";
-    surnameInput.value = "";
-    showLists();
-    addStudentButton.style.display = "block";
-    updateStudentButton.style.display = "none";
-    e.preventDefault();
+updateStudentButton.addEventListener("click", async (e) => {
+  await studentService.updateStudent(selectedStudent.id, nameInput.value, surnameInput.value);
+  nameInput.value = "";
+  surnameInput.value = "";
+  showLists();
+  addStudentButton.style.display = "block";
+  updateStudentButton.style.display = "none";
+  e.preventDefault();
 });
 
-takeAttendanceButton.addEventListener("click", (e) => {
-    const map = new Map();
-    studentService.getStudents().forEach((student: Student, index: number) => {
-        map.set(student, selectList[index].value === "+");
-    })
-    attendanceService.takeAttendance(selectPrayerTime.value, map);
-    console.log(attendanceService.getAttendanceList());
-    showLists();
-    listWithAttendanceId();
-    // listStudentsForAttendance();
-    e.preventDefault();
+takeAttendanceButton.addEventListener("click", async (e) => {
+  const map = new Map();
+ (await studentService.getStudents()).forEach((student: Student, index: number) => {
```

Comment:
Bazı servisler kullanımlarında await var ve asenkron olarak belirklenmişler. Halbuki bazılarında da yok. Bu await ve async kullanımı nerde ne için, nasıl oluyor öğrenip, sendeki hangi ihtiyaçtan kullandığına bakabilir misin?